vscode-dotnet-runtime-library/src/EventStream/TelemetryUtilities.ts

```diff
@@ -0,0 +1,17 @@
+ /* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+ import * as crypto from 'crypto';
+import { TextEncoder } from 'util';
+
+export class TelemetryUtilities
+{
+    public static HashData(dataToHash: string) : string

```

Comment:
nit: I notice in a few places that you have to ternary-check the data passed to this method in case it is null - could you instead change the signature here to `HashData(datatoHash: string?) : string` and have this method check for `null` and in that case return `''`? Might make your callsites less painful.

Reply:
Good point, thank you